client-core/src/service/hd_key_service.rs

```diff
@@ -16,9 +16,40 @@ const KEYSPACE: &str = "core_hd_key";
 struct HdKey {
     staking_index: u32,
     transfer_index: u32,
+    // curently only one viewkey per wallet, but it's good to keep it for uniformity.
+    viewkey_index: u32,
```

Comment:
because transfer_index & viewkey_index always the same,
how about use just one index?

Reply:
` staking_index `and  `transfer_index` are supposed to be the index for Staking address and Transfer address?

Reply:
yes,
so it's policy issue.
for ease of restoration addresses when you resotre hdwallet.

which do you think easier?

Reply:
Staking addresses and transfer addresses are independent, so `staking_index` and `transfer_index` will have different values.